lib/suggestion-list-element.coffee

```diff
@@ -162,30 +166,43 @@ class SuggestionListElement extends HTMLElement
     @descriptionContent = @querySelector('.suggestion-description-content')
     @descriptionMoreLink = @querySelector('.suggestion-description-more-link')
 
-  calculateMaxListHeight: ->
-    li = document.createElement('li')
-    li.textContent = 'test'
-    @ol.appendChild(li)
-    itemHeight = li.offsetHeight
-    paddingHeight = parseInt(getComputedStyle(this)['padding-top']) + parseInt(getComputedStyle(this)['padding-bottom']) ? 0
-    @scroller.style['max-height'] = "#{@maxVisibleSuggestions * itemHeight + paddingHeight}px"
-    li.remove()
-
   renderItems: ->
     items = @visibleItems() ? []
     @renderItem(item, index) for item, index in items
     li.remove() while li = @ol.childNodes[items.length]
-    @selectedLi?.scrollIntoView(false)
-
-    @descriptionContainer.style['max-width'] = "#{@offsetWidth}px"
+    @updateDescription()
 
     if @suggestionListFollows is 'Word'
-      firstChild = @ol.childNodes[0]
-      wordContainer = firstChild?.querySelector('.word-container')
-      marginLeft = 0
-      marginLeft = -wordContainer.offsetLeft if wordContainer?
+      wordContainer = @selectedLi?.querySelector('.word-container')
+      marginLeft = -(wordContainer?.offsetLeft ? 0)
       @style['margin-left'] = "#{marginLeft}px"
 
+  renderSelectedItem: ->
+    @selectedLi.classList.remove('selected')
+    @selectedLi = @ol.childNodes[@selectedIndex]
+    @selectedLi.classList.add('selected')
+    @scrollSelectedItemIntoView()
+    @updateDescription()
+
+  # This is reading the DOM in the updateDOM cycle. If we dont, there is a flicker :/
+  scrollSelectedItemIntoView: ->
+    scrollTop = @scroller.scrollTop
```

Comment:
Along similar lines, there are a bunch of reads here in the `updateDocument()` call to `renderSelectedItem`. This is not ideal, but prevents flickering. judging by the timeline, these ones are less bad than the `offsetLeft` read above.